File: irrgarten-java/src/irrgarten/Weapon.java
package irrgarten;

/**
 * Arma utilizada por el jugador durante el combate
 * @author vik
 */
public class Weapon {
    
    private float power;
    private int uses;

    public Weapon(float power, int uses) {
        this.power = power;
        this.uses = uses;
    }
    
    /**
     * Intensidad del ataque del jugador
     * @return 0 si uses == 0, power si uses > 0
     */
    public float attack() {
        float pow = 0;
        if (uses > 0) {
            pow = power;
            uses--;
        }
        return pow;
    }
    
    public boolean discard() {
        return Dice.discardElement(uses);
    }
    
    @Override
    public String toString() {
        return String.format("W[%f.2, %d]",power, uses);
    }
}

File: irrgarten-ruby/weapon.rb
# encoding: utf-8

require_relative 'dice'

module Irrgarten
  class Weapon
    def initialize(power, uses)
      @power = power
      @uses = uses
    end

    def attack
      out = 0
      if @uses > 0
        out = @power
        @uses -= 1
      end
      out
    end

    def to_s
      "W[#{@power}, #{@uses}]"
    end

    def discard
      Dice.discard_element(@uses)
    end
  end
end